pandas/core/frame.py

```diff
@@ -994,13 +1004,13 @@ def itertuples(self, index=True, name="Pandas"):
         # fallback to regular tuples
         return zip(*arrays)
 
-    def __len__(self):
+    def __len__(self) -> int:
         """
         Returns length of info axis, but here we use the index.
         """
         return len(self.index)
 
-    def dot(self, other):
+    def dot(self, other: Union[Series, DataFrame]) -> Union[Series, DataFrame]:
```

Comment:
Should include `np.ndarray` in the signature here. Applicable to the rest below as well

Reply:
Alright.
Added some types without paying much attention, will take a closer later.
Btw I am guessing if type of `other` is `Series`, return type will be `Series` too, similar for `DataFrame` and `np.ndarray`, how do I indicate that.

Reply:
A TypeVar would make the function generic (i.e. the return type would match `other` if the same TypeVar is used for both) but unfortunately I don't think that is how this is actually implemented

Reply:
hmm, pretty sure saw something like this in the docstring. But did this in a hurry(thats why draft PR), a bit busy. Will take a closer look as soon as I can.